ckan/config/middleware/flask_app.py

```diff
@@ -126,6 +126,29 @@ def _ungettext_alias():
     return dict(ungettext=ungettext)
 
 
+class BeakerSessionInterface(SessionInterface):
+    def open_session(self, app: Any, request: Any):
+        if 'beaker.session' in request.environ:
+            return request.environ['beaker.session']
+
+    def save_session(self, app: Any, session: Any, response: Any):
+        session.save()
+
+    def is_null_session(self, obj: object) -> bool:
+
+        is_null = super(BeakerSessionInterface, self).is_null_session(obj)
+
+        if not is_null:
+            # Beaker always adds these keys on each request, so if these are
+            # the only keys present we assume it's an empty session
+            is_null = (
+                sorted(obj.keys()) == [
```

Comment:
@smotornyuk can you help me out with the typing of this? I added the `object` type definition to the method because that what the parent Flask class uses but this gives the following [typing error](https://github.com/ckan/ckan/runs/7178711102?check_suite_focus=true#step:7:18):

```
   /home/runner/work/ckan/ckan/ckan/config/middleware/flask_app.py:145:28 - error: Cannot access member "keys" for type "object"
    Member "keys" is unknown (reportGeneralTypeIssues)
```

I tried various combinations of defining `obj` as a dict but failed, as my typing knowledge is limited

Reply:
Thanks a lot for the super-detailed explanation, I learnt a lot!